src/App.tsx

```diff
@@ -1,24 +1,148 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
+import { Todo } from './types/Todo';
+import { FilterOption } from './types/FilterOption';
+import { getTodos, createTodo, deleteTodo } from './api/todos';
+import { AddTodoInput } from './components/AddTodoInput';
+import { Footer } from './components/Footer';
+import { Alert } from './components/Alert';
 import { UserWarning } from './UserWarning';
+import { TodoList } from './components/TodoList';
+import { ErrorMessage } from './types/ErrorMessage';
 
-const USER_ID = 0;
+const USER_ID = 10527;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState(FilterOption.All);
+  const [hasError, setHasError] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [title, setTitle] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const handleAlert = (alertMessage: string) => {
+    setErrorMessage(alertMessage);
+    setHasError(true);
+
+    setTimeout(() => {
+      setHasError(false);
+      setErrorMessage('');
+    }, 3000);
+  };
+
+  const loadTodos = async () => {
+    try {
+      const todosFromServer = await getTodos(USER_ID);
+
+      setTodos(todosFromServer);
+    } catch {
+      handleAlert(ErrorMessage.Load);
+    }
+  };
+
+  const visibleTodos: Todo[] = useMemo(() => {
+    return todos.filter((todo) => {
+      switch (filter) {
+        case FilterOption.Completed:
+          return todo.completed;
+
+        case FilterOption.Active:
+          return !todo.completed;
+
+        default:
+          return true;
+      }
+    });
+  }, [todos, filter]);
+
+  const addTodo = async () => {
+    if (!title) {
+      handleAlert(ErrorMessage.EmptyTitle);
+
+      return;
+    }
+
+    try {
+      const todo = {
+        title,
+        id: 0,
```

Comment:
do not propagate `id` in create endpoint
```suggestion
```

Reply:
Ok your suggestion makes total sense to me. I'm not sure only because the task assignment mentions: "immediately after sending a request create a todo with id: 0 and save it to the tempTodo variable in the state (NOT to the todos array);" -> I can remove it, and then for Typescript do something like: "... createTodo = (todo: Omit<Todo, "id">) => ...". Is this what you'd like me to do or am I missing something else?